Update URL paths and remove unused code

amysorto opened the pull request:
Fixes #1243 

Also removed `LOI_JOB_ID_FRAGMENT_PARAM` since it not currently being used

gino-m commented on web/src/app/services/navigation/navigation.service.ts:
```diff
@@ -33,11 +33,10 @@ import {map} from 'rxjs/operators';
+  private static readonly JOB_ID_FRAGMENT_PARAM = 'job';
+  private static readonly LOI_ID_FRAGMENT_PARAM = 'site';
+  private static readonly SUBMISSION_ID_FRAGMENT_PARAM = 'submission';
+  private static readonly TASK_ID_FRAGMENT_PARAM = 'task';
```
Hi @amysorto ! This will rename the fragments, but we also need the router to use the URI rather than URL fragment. The exact patterns are described in the issue descriptions. Please lmk if you have any questions!

amysorto commented:
I deleted everything inside `pages/main-page-container/main-page/side-panel` folder as well since it wasn't being used anymore and used some of the code I deleted in the navigation service.

gino-m commented on web/src/app/routing.module.ts:
```diff
@@ -95,6 +95,21 @@ const routes: Routes = [
     canActivate: [AuthGuard],
   },
+  {
+    path: `${NavigationService.SURVEY_SEGMENT}/:${NavigationService.SURVEY_ID}/${NavigationService.LOI_SEGMENT}/:${NavigationService.LOI_ID}`,
```
Can we use destructuring assignment to make these constants more readable? ie somewhere above `const {SURVEY_SEGMENT, LOI_SEGMENT, ...} = NavigationService;`?

gino-m commented on web/src/app/services/navigation/navigation.service.ts:
```diff
@@ -182,86 +164,38 @@ export class NavigationService {
-    this.setFragmentParams(new HttpParams({fromString: ''}));
+  showSubmissionDetail(surveyId: string, loiId: string, submissionId: string) {
+    this.router.navigateByUrl(
+      `${NavigationService.SURVEY_SEGMENT}/${surveyId}/${NavigationService.LOI_SEGMENT}/${loiId}/${NavigationService.SUBMISSION_SEGMENT}/${submissionId}`
```
Same here.

gino-m reviewed (changes requested):
Some quick feedback, will do another pass tomorrow.

gino-m commented on web/src/app/services/navigation/navigation.service.ts:
```diff
@@ -186,7 +180,7 @@ export class NavigationService {
   ) {
     this.router.navigateByUrl(
-      `${NavigationService.SURVEY_SEGMENT}/${surveyId}/${NavigationService.LOI_SEGMENT}/${loiId}/${NavigationService.SUBMISSION_SEGMENT}/${submissionId}/${NavigationService.TASK_SEGMENT}/${taskId}`
+      `${SURVEY_SEGMENT}/${surveyId}/${LOI_SEGMENT}/${loiId}/${SUBMISSION_SEGMENT}/${submissionId}/${TASK_SEGMENT}/${taskId}`
```
FYI: Using taskId to identify the result will become problematic once we allow repeated tasks. That's out of scope for this PR, just wanted to call it out for future changes.

amysorto replied:
That's good to know! Currently the taskId is used just to highlight the task

gino-m reviewed (approved):
LGTM! Have you tests all routes manually?
